research/global-config/index.md

```diff
@@ -0,0 +1,361 @@
+
+You can configure services for individual projects in two ways:
+
+1. Top-Down solution: configuration resides on the service side, managed by the service team. Users notify the team to opt in or out. Example: Zuul for Fedora CI uses this approach ([configuration reference](https://pagure.io/fedora-project-config/blob/master/f/resources/fedora-distgits.yaml)).
+
+2. Bottom-Up solution: configuration lives in the user's project repository. Packit currently implements this approach.
+
+The top-down approach represents a global configuration, while the bottom-up solution can reference global or semi-global configurations. Let's analyze both approaches.
+
+# Top-Down solution
+
+The service side maintains project-specific behavior configurations.
+
+## Advantages
+
+- Users don't need configuration files in their projects; they might not like the idea of having a service related file in their projects.
+- Configuration migrations are straightforward for the service team; it is just a change in a file that belongs to the team.
+- Highest performance due to direct configuration access; no need to load and pre-process other files.
+- Quickest implementation; this approach is the same as for the packit-service configuration file.
+
+## Disadvantages
+
+- It is probably easier, for the final user, to look for the service configuration file in its own repo instead of a service repo.
+- Less user engagement due to limited visibility of configuration changes. Even though a configuration migration can be simpler for the service team it could be less explicit for the final user. Being able to change behaviour without the user acknowledging it could not be a good idea. In packit there already is a configuration migration script.
+- The global configuration file could really grow huge and be difficult to maintain both for the service team and for a final user that wants to contribute to it.
+- Lacks ecosystem-specific configuration management. There is no encapsulation for "middle layer" knowledge, no easy way to manage ecosystem configurations. Configurations could be grouped in different nodes with different defaults but there is no easy way, for the service team, to know if the user who is asking to update a configuration has the rights for doing so.
+- Differs from Packit's current approach, potentially confusing users. This could work mainly for the _Fedora CI_ and could confuse users that use the standard packit configuration for _upstream continuous integration_ and _release synchronization_.
+
+# Bottom-Up solution
+
+Project-side configuration with ability to reference global configurations.
+
+## Advantages
+
+- User-friendly and explicit. Everything is on the user side (more quick for him) and the file will never grow too much because of other project details.
+- Flexible customization through layered configurations and better ecosystem-specific configuration support.
```

Comment:
Big :+1:  for this. It would be really nice to get SIGs help in this. And maybe create a shared space for putting these together.